src/scheduler/ExecGraph.cpp

```diff
@@ -25,6 +25,28 @@ int countExecGraphNodes(const ExecGraph& graph)
     return count;
 }
 
+std::set<std::string> getExecGraphHostsForNode(const ExecGraphNode& node)
+{
+    std::set<std::string> hostsForNode;
+    hostsForNode.insert(node.msg.executedhost());
+
+    if (!node.children.empty()) {
+        for (auto c : node.children) {
+            std::set<std::string> nodeHost = getExecGraphHostsForNode(c);
+            hostsForNode.insert(nodeHost.begin(), nodeHost.end());
+        }
+    }
+
+    return hostsForNode;
+}
+
+std::set<std::string> getExecGraphHosts(const ExecGraph& graph)
+{
+    ExecGraphNode rootNode = graph.rootNode;
+    std::set<std::string> hosts = getExecGraphHostsForNode(rootNode);
+    return hosts;
```

Comment:
Very very nitpicky but if you're changing things I'd say this would be just as clear with:

```cpp
return getExecGraphHostsForNode(graph.rootNode)
```